### ISSUE TITLE
[Feature] Disable clustering support at compile time 

### ISSUE THREAD

**Issue by s3rj1k** [lxc/lxd#6145]
Would be nice to have an option to disable clustering support at compile time, so that one would not need to build custom patched sqlite.

This will help greatly with LXD Debian packaging.

**Comment by stgraber** [comment]
We will not be doing this. Doing so would require supporting two entirely different database backends in parallel as well as having part of the API shut off based on build tags. That would double our testing efforts, our maintenance efforts and introduce a lot of complexity in already complex code.
